File: modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/model/PriceEntry.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.pricing.web.internal.model;

import com.liferay.commerce.frontend.model.ImageField;

/**
 * @author Alessio Antonio Rendina
 */
public class PriceEntry {

	public PriceEntry(
		String basePrice, ImageField image, String name, long priceEntryId,
		String sku, String tieredPrice, String unitDiscount, String unitPrice) {

		_basePrice = basePrice;
		_image = image;
		_name = name;
		_priceEntryId = priceEntryId;
		_sku = sku;
		_tieredPrice = tieredPrice;
		_unitDiscount = unitDiscount;
		_unitPrice = unitPrice;
	}

	public String getBasePrice() {
		return _basePrice;
	}

	public ImageField getImage() {
		return _image;
	}

	public String getName() {
		return _name;
	}

	public long getPriceEntryId() {
		return _priceEntryId;
	}

	public String getSku() {
		return _sku;
	}

	public String getTieredPrice() {
		return _tieredPrice;
	}

	public String getUnitDiscount() {
		return _unitDiscount;
	}

	public String getUnitPrice() {
		return _unitPrice;
	}

	private final String _basePrice;
	private final ImageField _image;
	private final String _name;
	private final long _priceEntryId;
	private final String _sku;
	private final String _tieredPrice;
	private final String _unitDiscount;
	private final String _unitPrice;

}

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/ServiceProvider/commerce-admin-pricing/v2.0/PriceEntry.js
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import AJAX from '../../../utilities/AJAX/index';

const VERSION = 'v2.0';

function resolvePath(basePath, priceListId, priceEntryId) {
	let path = `${basePath}${VERSION}/price-lists/${priceListId}/price-entries`;

	if (priceEntryId) {
		path += `/${priceEntryId}`;
	}

	return path;
}

export default (basePath) => ({
	addPriceEntry: (priceListId, json) =>
		AJAX.POST(resolvePath(basePath, priceListId), json),
});
